consensus/hotstuff/distributor.go

```diff
@@ -0,0 +1,30 @@
+package hotstuff
+
+import "github.com/onflow/flow-go/consensus/hotstuff/model"
+
+// Distributor is an interface for distributing HotStuff consensus events to registered consumers.
+// It provides a pub-sub mechanism where components can register callbacks to be notified
+// when blocks are finalized or incorporated into the chain.
+//
+// The distributor pattern centralizes event distribution and ensures that all registered
+// consumers are notified of consensus events. This prevents callers from forgetting to
+// register callbacks and provides a consistent way to handle consensus notifications.
+type Distributor interface {
```

Comment:
```suggestion
// FinalizationRegistrar is an interface for registering consumers of HotStuff events.
// It provides a pub-sub mechanism where components can register callbacks to be notified
// when blocks are finalized or incorporated into the chain.
//
// Components which must be subscribed to finalization events for correct functionality should:
//    - accept a FinalizationRegistrar argument in their constructor
//    - register their event consumer functions within their constructor
// Following this pattern of coupling construction with subscription prevents callers from forgetting
// to register callbacks and provides a consistent way to handle consensus notifications.
type FinalizationRegistrar interface {
```
Suggesting to change the name: `FinalizationRegistrar` because we're only working with finalization events, and this interface is registering consumers only (not actually distributing events). Also adjusting the godoc to be explicit about what pattern we need users to follow (the interface on its own is insufficient -- we need components to make use of it in a specific way, by subscribing in their constructors).